examples/tree_1d_dgsem/elixir_shallowwater_wall.jl

```diff
@@ -0,0 +1,78 @@
+
```

Comment:
This elixir is never tested. Although I think we can just remove it from the current PR and move it into different PR.

This problem contains shocks so it would be good to add the capability to have shock capturing for 1D shallow water. This requires that the `calcflux_fv!` is adapted to accept surface fluxes that have nonconservative terms. This means that some of the caches and call structure of `calcflux_fv!` needs changed to accept left/right arguments. More importantly, for the FV shallow water scheme to maintain well-balancedness, special surface/nonconservative terms that perform a hydrostatic reconstruction on the water height is necessary. This second fact is needed for the 2D version as well.

I have worked out how to do all these things. But I think it fits better in a different PR (after this one is merged) because it touches the 2D implementation too. Does that plan sound reasonable @ranocha @sloede ?

Reply:
I will remove the file from the PR and save it locally on my PC.
If you will need it in the future, just let me know.

Reply:
Sounds good

Reply:
Yes, this sounds good to me :+1: I always prefer reviewing small PRs focusing on a single aspect instead of a giant amount of code changes